readme_renderer/markdown.py

```diff
@@ -61,7 +66,8 @@ def render(raw, variant="GFM", **kwargs):
     if not renderer:
         return None
 
-    rendered = renderer(raw)
+    # The renderer is a lambda function, and mypy fails lambdas right now.
+    rendered = renderer(raw)  # type: ignore
```

Comment:
Did you try to work around this? I wonder if [`Callable`](https://mypy.readthedocs.io/en/stable/kinds_of_types.html#callable-types-and-lambdas) and/or [`cast`](https://mypy.readthedocs.io/en/stable/cheat_sheet_py3.html#when-you-re-puzzled-or-when-things-are-complicated) would help. Or, maybe `variants` could be updated to use explicit private functions with annotations, instead of lambdas.

Reply:
I did not try to modify the code to make the type annotations work - that felt counter to the effort at hand. mypy struggles with lambdas - see open issue: https://github.com/python/mypy/issues/4226

It'd be awesome if you wanted to take on that as a follow-up once this is merged to remove the ignore!